modules/components/org.wso2.carbon.appfactory.core/src/main/java/org/wso2/carbon/appfactory/core/workflow/WorkflowExecutorFactory.java

```diff
@@ -0,0 +1,132 @@
+import org.wso2.carbon.appfactory.core.workflow.dto.TenantCreationWorkflowDTO;
+import org.wso2.carbon.appfactory.core.workflow.dto.WorkflowDTO;
+
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+
+/**
+ * The factory class is creating workflow dto object and workflow executor object
+ */
+public class WorkflowExecutorFactory {
+
+    private static final WorkflowExecutorFactory workflowExecutorFactory = new WorkflowExecutorFactory();
+    private static Log log = LogFactory.getLog(WorkflowExecutorFactory.class);
+    private Map<WorkflowConstant.WorkflowType, WorkflowExecutor> workflowExecutorMap;
+
+    private WorkflowExecutorFactory() {
+
+        try {
+            loadWorkflowConfiguration();
+        } catch (AppFactoryException e) {
+            String message = "Unable to load workflow configuration or create workflow objects";
+            log.error(message, e);
+        }
+
+    }
+
+    public static WorkflowExecutorFactory getInstance() {
+        return workflowExecutorFactory;
+    }
+
+    /**
+     * Load configured class to the class loader and
+     * create instance of the particular class
+     */
+    public void loadWorkflowConfiguration() throws AppFactoryException {
+        if (log.isDebugEnabled()) {
+            String message = "Load configured class to class loader and create instance of that particular class "
+                    + "dynamically";
+            log.debug(message);
+        }
+
+        workflowExecutorMap = new ConcurrentHashMap<WorkflowConstant.WorkflowType, WorkflowExecutor>();
+        String executorClass;
+        WorkflowExecutor workflowExecutor;
+        try {
+            executorClass = AppFactoryUtil.getAppfactoryConfiguration()
+                    .getFirstProperty(WorkflowConstant.TENANT_CREATION);
+        } catch (AppFactoryException e) {
+            String message = "Unable to read executor class from the appfactory.xml configuration";
+            throw new AppFactoryException(message, e);
+        }
+
+        if (StringUtils.isBlank(executorClass)) {
+            executorClass = WorkflowConstant.TENANT_CREATION_DEFAULT_WORKFLOW_CLASS;
+            String message = "The tenant creation executor class is not define in appfactory.xml because tenant "
+                    + "creation default workflow is " + "executing. The tenant creation default workflow class is : "
+                    + executorClass;
+            log.warn(message);
+        }
+
+        try {
+            Class tenantCreationExecutorClass = WorkflowExecutorFactory.class.getClassLoader().loadClass(executorClass);
+            workflowExecutor = (WorkflowExecutor) tenantCreationExecutorClass.newInstance();
+            workflowExecutorMap.put(WorkflowConstant.WorkflowType.TENANT_CREATION, workflowExecutor);
+        } catch (ClassNotFoundException e) {
+            String message = "Unable to find configured class from the source code : class name : " + executorClass;
```

Comment:
Format the log message in more user friendly manner.
ex:
"Unable to find the class: " + executorClass + " in the class loader , when initializing the executor class for workflow type: " + WorkflowConstant.WorkflowType.TENANT_CREATION;